lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +23,147 @@ def fragment_cache(cached_hash, non_cached_hash)
           Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
         end
 
+        private
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # nested_associations is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def add_resource_relationships(parent, serializer, nested_associations = [])
+          # have the include array normalized
+          nested_associations = include_array_to_hash(nested_associations)
+
+          included_associations = if nested_associations.present?
+            serializer.associations.select{ |association|
+              # nested_associations is a hash of:
+              #   key => nested association to include
+              nested_associations.keys.include?(association.name)
+            }
+          else
+            serializer.associations
+          end
+
+          included_associations.each do |association|
```

Comment:
I think you can loop straight through `serializer.associations` here and  delete the whole `included_associations` block before it because of [what I mentioned](https://github.com/rails-api/active_model_serializers/pull/1114/files#r38724815)